cmake/CMakeLists.txt

```diff
@@ -698,14 +699,17 @@ if(NOT onnxruntime_USE_FULL_PROTOBUF)
 else()
   set(ONNX_USE_LITE_PROTO OFF CACHE BOOL "" FORCE)
 endif()
-add_subdirectory(external/onnx)
-target_compile_definitions(onnx PUBLIC $<TARGET_PROPERTY:onnx_proto,INTERFACE_COMPILE_DEFINITIONS> PRIVATE "__ONNX_DISABLE_STATIC_REGISTRATION")
-if (NOT onnxruntime_USE_FULL_PROTOBUF)
-  target_compile_definitions(onnx PUBLIC "__ONNX_NO_DOC_STRINGS")
+if (NOT onnxruntime_MINIMAL_BUILD)
+  add_subdirectory(external/onnx)
+  target_compile_definitions(onnx PUBLIC $<TARGET_PROPERTY:onnx_proto,INTERFACE_COMPILE_DEFINITIONS> PRIVATE "__ONNX_DISABLE_STATIC_REGISTRATION")
+  if (NOT onnxruntime_USE_FULL_PROTOBUF)
+    target_compile_definitions(onnx PUBLIC "__ONNX_NO_DOC_STRINGS")
+  endif()
+  set_target_properties(onnx PROPERTIES FOLDER "External/ONNX")
+  set_target_properties(onnx_proto PROPERTIES FOLDER "External/ONNX")
+else()
+  include(onnx_minimal)
 endif()
-set_target_properties(onnx PROPERTIES FOLDER "External/ONNX")
-set_target_properties(onnx_proto PROPERTIES FOLDER "External/ONNX")
-
 if (onnxruntime_USE_MKLML)
```

Comment:
Will refactor this a little as the calls to target_compile_definitions and set_target_properties are duplicated in onnx_minimal.cmake.